lib/charms/observability_libs/v1/kubernetes_service_patch.py

```diff
@@ -160,6 +179,7 @@ def __init__(
         ports: List[ServicePort],
         service_name: Optional[str] = None,
         service_type: ServiceType = "ClusterIP",
+        is_new_service: Optional[bool] = False,
```

Comment:
(ignore me if this has been discussed elsewhere) this feels like adding a new concern to `KubernetesServicePatch`.  That KubernetesService**Patch** creates a new service feels odd, especially since the docstring says """A utility for patching the Kubernetes service set up by Juju."  

What about using something standalone for creating new services?

Reply:
Currently, we have `ServiceType = Literal["ClusterIP", "LoadBalancer"]`, but we no longer use this lib for the "ClusterIP" use case, because we have ops open_port.

That being said, at first glance it seems to me that we can remove the `is_new_service` arg entirely, and always create a new service for LoadBalancer (but keep patching if ClusterIP).
This way we maintain backwards compat, while solving the LoadBalancer issue with no additional code changes beyond fetch-lib.
Wdyt?

Reply:
Thanks this definitely looks a lot better, it should be updated now

Reply:
To me it still feels odd that something labeled "patch" creates new things - I don't think someone coming to this lib would expect the behaviour.  But I also respect this is a low-code way of addressing the issue for traefik.